bazel/cel-cpp.patch

```diff
@@ -0,0 +1,1360 @@
+ cc_test(
+     name = "converters_test",
+     srcs = ["converters_test.cc"],
+-    copts = ["-std=c++14"],
++    copts = ["-std=c++17"],
+     data = [
+         "@com_google_cel_spec//testdata",
+     ],
+diff --git a/testutil/BUILD b/testutil/BUILD
+index 0bf8e50..7305225 100644
+--- a/testutil/BUILD
++++ b/testutil/BUILD
+@@ -15,7 +15,7 @@ cc_library(
+     hdrs = [
+         "test_data_util.h",
+     ],
+-    copts = ["-std=c++14"],
++    copts = ["-std=c++17"],
+     deps = [
+         "//common:type",
+         "//common:value",
+@@ -44,7 +44,7 @@ cc_library(
+     hdrs = [
+         "test_data_io.h",
+     ],
+-    copts = ["-std=c++14"],
++    copts = ["-std=c++17"],
+     deps = [
+         "//internal:status_util",
+         "@com_github_google_googletest//:gtest_main",
+@@ -69,7 +69,7 @@ cc_binary(
+     srcs = [
+         "test_data_gen.cc",
+     ],
+-    copts = ["-std=c++14"],
++    copts = ["-std=c++17"],
+     deps = [
+         ":test_data_io",
+         ":test_data_util",
+@@ -87,7 +87,7 @@ cc_binary(
+ cc_test(
+     name = "test_data_test",
+     srcs = ["test_data_test.cc"],
+-    copts = ["-std=c++14"],
++    copts = ["-std=c++17"],
+     data = [
+         "@com_google_cel_spec//testdata",
+     ],
+@@ -105,7 +105,7 @@ cc_library(
+     hdrs = [
+         "util.h",
+     ],
+-    copts = ["-std=c++14"],
++    copts = ["-std=c++17"],
+     deps = [
+         "@com_github_google_googletest//:gtest_main",
+         "@com_google_protobuf//:protobuf",
+diff --git a/tools/BUILD b/tools/BUILD
+index 212207e..8a131bd 100644
+--- a/tools/BUILD
++++ b/tools/BUILD
+@@ -17,7 +17,7 @@ cc_library(
+     hdrs = [
+         "flatbuffers_backed_impl.h",
+     ],
+-    copts = ["-std=c++14"],
++    copts = ["-std=c++17"],
+     deps = [
+         "//eval/public:cel_value",
+         "@com_github_google_flatbuffers//:flatbuffers",
+@@ -36,7 +36,7 @@ cc_library(
+     name = "flatbuffers_test_cc",
+     srcs = [":flatbuffers_test"],
+     hdrs = [":flatbuffers_test"],
+-    copts = ["-std=c++14"],
++    copts = ["-std=c++17"],
+     features = ["-parse_headers"],
+     linkstatic = True,
+     deps = ["@com_github_google_flatbuffers//:runtime_cc"],
+@@ -48,7 +48,7 @@ cc_test(
+     srcs = [
+         "flatbuffers_backed_impl_test.cc",
+     ],
+-    copts = ["-std=c++14"],
++    copts = ["-std=c++17"],
+     data = [
+         ":flatbuffers_reflection_out",
+     ],
+diff --git a/v1beta1/BUILD b/v1beta1/BUILD
+index 41c64d6..2480b94 100644
+--- a/v1beta1/BUILD
++++ b/v1beta1/BUILD
+@@ -15,7 +15,7 @@ cc_library(
+     hdrs = [
+         "converters.h",
+     ],
+-    copts = ["-std=c++14"],
++    copts = ["-std=c++17"],
+     deps = [
+         "//common:converters",
+         "//common:macros",
+@@ -37,7 +37,7 @@ cc_library(
+ cc_test(
+     name = "converters_test",
+     srcs = ["converters_test.cc"],
+-    copts = ["-std=c++14"],
++    copts = ["-std=c++17"],
+     data = [
+         "@com_google_cel_spec//testdata",
+     ],
```

Comment:
This PR looks good, but I'm not super happy with this patch. Can we fix this upstream? Or, alternatively, can we `sed` this rather than patching? I'm generally opposed to any patch applied to external dependencies as they are a maintenance nightmare and hurt our ability to upgrade dependencies by creating toil.